tests/unit/test_handlers.py

```diff
@@ -332,6 +332,50 @@ def test_get_template_has_error_response(self):
         # an error response.
         self.assertEqual(original, handler_input)
 
+    def test_encode_json_policy(self):
+        params = {
+            'Document': {
+                'foo': 'bar'
+            },
+            'Other': 'baz'
+        }
+        service_def = {
```

Comment:
You should try to use the [DenormalizedStructureBuilder](https://github.com/boto/botocore/blob/develop/botocore/model.py#L487). It is a lot easier for creating shapes.